### PR TITLE
[5.3] Fix crash related to build conditionals

### PR DESCRIPTION
**Pull request by neonichu**
This fixes a crash related to build conditionals which looks like a
longer standing bug that was triggered by https://github.com/apple/swift-llbuild/pull/681
Unforunately, llbuild CI doesn't test the package, so it slipped through
and now causes SwiftPM to crash when building itself

rdar://problem/67432513

### CODE REVIEW

**Comment by neonichu** [comment]
This is a cherry-pick of the second part of #2876, but not the other part since that isn't appropriate for 5.3

**Comment by neonichu** [comment]
@swift-ci please test

**Review by abertelrud — APPROVED** [review]

**Review by abertelrud — COMMENTED** [review]
Looks good to me, though this will need release manager approval (didn’t initially see that it was targeted at 5.3).

**Review by rballard — APPROVED** [review]
Approved to merge; thank you!

**Pull request merged**
